lib/src/scope.dart

```diff
@@ -48,179 +44,56 @@ class WizardScopeState extends State<WizardScope> {
-    Object? arguments,
-    WizardRouteCallback? advance,
-  ) {
-    final routes = _getRoutes();
-    assert(routes.isNotEmpty, routes.length.toString());
-
-    final previous = WizardRouteSettings(
-      name: routes.last.name,
-      arguments: arguments,
-    );
-
-    // advance to a specific route
-    String? onNext() => advance?.call(previous);
-
-    // pick the next route on the list
-    String nextRoute() {
-      final index = widget._routes.indexOf(previous.name!);
-      assert(index < widget._routes.length - 1,
-          '`Wizard.next()` called from the last route ${previous.name}.');
-      return widget._routes[index + 1];
-    }
-
-    final name = onNext() ?? nextRoute();
-    assert(widget._routes.contains(name),
-        '`Wizard.routes` is missing route \'$name\'.');
-
-    return WizardRouteSettings<T?>(name: name, arguments: arguments);
-  }
+  Future<T?> next<T extends Object?>({T? arguments}) =>
+      widget._controller.next(arguments: arguments);
 
   /// Requests the wizard to replace the current page with the next one.
   /// Optionally, `arguments` can be passed to the next page.
   ///
   /// ```dart
   /// onPressed: () => Wizard.of(context).replace(arguments: something),
   /// ```
-  void replace({Object? arguments}) async {
-    final next = _getNextRoute(arguments, widget._route.onReplace);
-
-    _updateRoutes((state) {
-      final copy = List<WizardRouteSettings>.of(state);
-      copy[copy.length - 1] = next;
-      return copy;
-    });
-  }
+  void replace({Object? arguments}) =>
+      widget._controller.replace(arguments: arguments);
 
   /// Requests the wizard to jump to a specific page. Optionally, `arguments`
   /// can be passed to the page.
-  void jump(String route, {Object? arguments}) async {
-    assert(widget._routes.contains(route),
-        '`Wizard.jump()` called with an unknown route $route.');
-    final settings = WizardRouteSettings(name: route, arguments: arguments);
-
-    _updateRoutes((state) {
-      final copy = List<WizardRouteSettings>.of(state);
-      return copy..add(settings);
-    });
-  }
-
-  List<WizardRouteSettings> _getRoutes() =>
-      context.flow<List<WizardRouteSettings>>().state;
+  void jump(String route, {Object? arguments}) =>
+      widget._controller.jump(route, arguments: arguments);
 
-  void _updateRoutes(
-    List<WizardRouteSettings> Function(List<WizardRouteSettings>) callback,
-  ) {
-    context.flow<List<WizardRouteSettings>>().update(callback);
-  }
+  List<WizardRouteSettings> _getRoutes() => widget._controller.state;
 
   /// Returns `false` if the wizard page is the first page.
   bool get hasPrevious => widget._index > 0;
 
   /// Returns `false` if the wizard page is the last page.
   bool get hasNext {
-    if (widget._routes.isEmpty) return false;
+    final routes = widget._controller.routes;
+    if (routes.isEmpty) return false;
     final previous = _getRoutes().last.name!;
```

Comment:
could use controller.currentRoute? would that allow use to get rid of _getRoutes() altogether or is it sill used somewhere else?